### ISSUE TITLE
Core/Movement Loop in WaypointMovementGenerator<Creature>::OnArrived

### ISSUE THREAD

**Issue by sirikfoll** [TrinityCore/TrinityCore#18594]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:** There is a loop that keeps calling the method [WaypointMovementGenerator<Creature>::OnArrived](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp#L72) with the same WaypointID after it has been reached.
It can be seen in every creature that uses GetMotionMaster()->MovePath() and the MovementInform hook.
@Riztazz this could be of your interest

**Current behaviour:** Loop in WaypointMovementGenerator<Creature>::OnArrived with the same Waypoint ID.

**Expected behaviour:** Only 1 call for each Waypoint ID.

**Steps to reproduce the problem:**

1. .go crea id 26693
2. Start the fight with skadi;
3. Wait till he gets in range of the harpoons for the second time;
4. Note a spam from the emote [here](https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/UtgardeKeep/UtgardePinnacle/boss_skadi.cpp#L389) , because there will be a loop in MovementInform with WaypointID 11, note that he doesn't perform the me->SetFacing that is supposed to be done when reaching the waypoint.

**Branch(es):** 3.3.5

**TC rev. hash/commit:**  e8314e2c4aa6

**Comment by xinef1** [comment]
Try reverting 28050f338dfc66e0c40b6a3915bf96e38e0613e5

**Comment by sirikfoll** [comment]
This fixes it, but I don't know if breaks the smooth movement
```diff
diff --git a/src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp b/src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp
index 53ff5a9..71f0f00 100755
--- a/src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp
+++ b/src/server/game/Movement/MovementGenerators/WaypointMovementGenerator.cpp
@@ -156,6 +156,8 @@ bool WaypointMovementGenerator<Creature>::StartMove(Creature* creature)
                     transportPath = false;
                 // else if (vehicle) - this should never happen, vehicle offsets are const
             }
+
+            creature->GetMotionMaster()->Initialize();
             return false;
         }

**Comment by xinef1** [comment]
I took a deeper look. I don't know if you are aware have the motion master works, but the problem here comes from putting the waypoint movement on idle slot. When he reaches last point the waypoint movement generator is stuck in idle slot and when there are no other higher priority movement generators (active, controlled) it updates the waypoint gen all the time which results in continuous MovementInform calls because he reached this point. What you did drop all movement generators and puts Idle generator in idle slot, thats why it works properly. It shouldn't break anything but beware that this line will delete this object.

**Comment by irgc** [comment]
cunfirm

**Comment by akrom23** [comment]
It's quite strange, because ``return false`` there should remove the movement generator (``StartMove`` is called in ``StartMoveNow``, ``StartMoveNow`` is called in waypoint move genenerator's update).

**Comment by Keader** [comment]
Closed in a3c6880579f3326088ecbe5b8c08c4b75ed91a59
